src/store/item-store.js

```diff
@@ -55,4 +55,14 @@ store.registerHandler('STUDENT_FOUND', data => {
     }
     store.emitChange();
 });
+
+store.registerHandler('ITEM_DUEDATE_UPDATED', item => {
+    for(let i = 0; i < items.length; i++){
```

Comment:
array.prototype.find() is a bit more efficient, since it will return once the item is found.  If there are thousands of items in the system, it could end up saving a bit of time.  (It's also a little nicer looking)